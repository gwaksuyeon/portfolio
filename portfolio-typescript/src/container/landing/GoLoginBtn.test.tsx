import React from 'react';
import { Router } from 'react-router-dom'
import { render, fireEvent } from '@testing-library/react'
import { createMemoryHistory } from 'history'
import GoLoginBtnContainer from './GoLoginBtn';

const renderWithRouter = (component: React.ReactNode) => {
  const history = createMemoryHistory()
  return { 
  ...render (
  <Router history={history}>
      {component}
  </Router>
  )
}
}

describe('<GoLoginBtnContainer />', () => {
  it('matches snapshot', () => {
    const utils = renderWithRouter(<GoLoginBtnContainer text="Login" />);
    expect(utils.container).toMatchSnapshot();
  });

  it('has a text', () => {
    const utils = renderWithRouter(<GoLoginBtnContainer text="Login" />);
    utils.getByText('Login');
  });

  it('onClick', () => {
    const utils = renderWithRouter(<GoLoginBtnContainer text="Login" />);
    const text = utils.getByText('Login');

    fireEvent.click(text);
  })
});