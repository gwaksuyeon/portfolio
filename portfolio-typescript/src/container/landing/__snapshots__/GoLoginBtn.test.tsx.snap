// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GoLoginBtnContainer /> matches snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Login
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
